tests/common/member_test.go

```diff
@@ -52,3 +55,157 @@ func TestMemberList(t *testing.T) {
+					var err error
+					if tc.learner {
+						addResp, err = cc.MemberAddAsLearner(ctx, "newmember", []string{"http://localhost:123"})
+					} else {
+						addResp, err = cc.MemberAdd(ctx, "newmember", []string{"http://localhost:123"})
+					}
+					if nc.config.ClusterSize == 1 {
+						checkAddResp(t, addResp, err)
+					} else {
+						if err == nil {
+							t.Fatalf("expect MemberAdd to fail, got no error, addResp: %v", addResp)
+						}
+						if !strings.Contains(err.Error(), "etcdserver: unhealthy cluster") {
+							t.Fatalf("expected: 'etcdserver: unhealthy cluster' error,  got: %s", err)
+						}
+					}
+				})
+			})
+		}
+	}
+}
+
+func TestMemberAdd_SleetHealthInterval(t *testing.T) {
+	testRunner.BeforeTest(t)
+
+	tcs := []struct {
+		name    string
+		learner bool
+	}{
+		{
+			name:    "NotLearner",
+			learner: false,
+		},
+		{
+			name:    "Learner",
+			learner: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		for _, nc := range clusterTestCases {
+			t.Run(tc.name+"/"+nc.name, func(t *testing.T) {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+				clus := testRunner.NewCluster(ctx, t, nc.config)
+				defer clus.Close()
+				cc := clus.Client()
+
+				testutils.ExecuteUntil(ctx, t, func() {
+					clus.WaitLeader(t)
+					var addResp *clientv3.MemberAddResponse
+					var err error
+					time.Sleep(etcdserver.HealthInterval)
+					if tc.learner {
+						addResp, err = cc.MemberAddAsLearner(ctx, "newmember", []string{"http://localhost:123"})
+					} else {
+						addResp, err = cc.MemberAdd(ctx, "newmember", []string{"http://localhost:123"})
+					}
+					checkAddResp(t, addResp, err)
+				})
+			})
+		}
+	}
+}
+
+func TestMemberAdd_DisableStrictReconfigCheck(t *testing.T) {
+	testRunner.BeforeTest(t)
+
+	tcs := []struct {
+		name    string
+		learner bool
+	}{
+		{
+			name:    "NotLearner",
+			learner: false,
+		},
+		{
+			name:    "Learner",
+			learner: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		for _, strictNc := range clusterTestCases {
+			// make a copy of strictNc and disable StrictReconfigCheck for it
+			nc := strictNc
+			nc.config.DisableStrictReconfigCheck = true
+			t.Run(tc.name+"/"+nc.name, func(t *testing.T) {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+				clus := testRunner.NewCluster(ctx, t, nc.config)
+				defer clus.Close()
+				cc := clus.Client()
+
+				testutils.ExecuteUntil(ctx, t, func() {
+					clus.WaitLeader(t)
+					var addResp *clientv3.MemberAddResponse
+					var err error
+					if tc.learner {
+						addResp, err = cc.MemberAddAsLearner(ctx, "newmember", []string{"http://localhost:123"})
+					} else {
+						addResp, err = cc.MemberAdd(ctx, "newmember", []string{"http://localhost:123"})
+					}
+					checkAddResp(t, addResp, err)
+				})
+			})
+		}
+	}
+}
+
+func checkAddResp(t *testing.T, addResp *clientv3.MemberAddResponse, err error) {
```

Comment:
This checks error and validates response. Could we maybe split this:
* rename `checkAddResp` to `validateMemberAdd` and remove `err` argument.
* for error checking we can use `require.NoError(err)`.

Reply:
Thanks to the suggestions! I'd love to do so , as they make the code clearer.